File: vulntest/tier1-single/stack-overflow/csharp/remediation/Vuln.cs
// Tier 1 — stack-overflow / C# — remediation.
//
// Stay out of `unsafe`. Use Span<byte> stackalloc form which is
// bounds-checked, or System.Buffers / managed arrays.

using System;

class Vuln {
    static void Greet(string name) {
        Span<byte> buf = stackalloc byte[64];
        int n = Math.Min(name.Length, buf.Length);
        for (int i = 0; i < n; i++) buf[i] = (byte)name[i];   // span access bounds-checked
    }

    static int Main(string[] args) {
        if (args.Length < 1) return 1;
        Greet(args[0]);
        return 0;
    }
}

File: vulntest/tier1-single/stack-overflow/cpp/remediation/vuln.cpp
/*
 * Tier 1 — stack-overflow / C++ — remediation.
 *
 * Idiomatic C++ fix: replace the C-style char buffer with std::string,
 * which grows dynamically and bounds itself. Alternative: std::setw()
 * on operator>> caps input length at the buffer size.
 */
#include <iostream>
#include <string>

class Greeter {
public:
    void promptAndGreet() {
        std::string name;            // grows as needed; no fixed bound to overflow
        std::cout << "Name: ";
        std::cin >> name;
        std::cout << "Hello, " << name << "\n";
    }
};

int main() {
    Greeter g;
    g.promptAndGreet();
    return 0;
}
